packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_request_wrapper.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/foundation.dart';
+import 'package:in_app_purchase/src/in_app_purchase_connection/product.dart';
+
+const MethodChannel _channel =
+    MethodChannel('plugins.flutter.io/in_app_purchase');
+
+/// [SKProductRequestWrapper] wraps IOS SKProductRequest class to to retrive StoreKit product information in dart.
+///
+/// https://developer.apple.com/documentation/storekit/skproductsrequest?language=objc
+class SKProductRequestWrapper {
+  /// Get product list.
+  ///
+  /// [identifiers] is the product identifiers specified in Itunes Connect for the products that need to be retrived.
+  ///
+  /// Returns a future containing a list of [SKProduct] which then can be queried to get desired information.
+  static Future<List<Product>> getProductList(List<String> identifiers) async {
```

Comment:
So in the original API the request can be started and canceled. Do you think that's worth keeping here? The Dart API equivalent is probably [`CancelableOperation<T>`](https://pub.dartlang.org/documentation/async/latest/async/CancelableOperation-class.html). You'd need iOS to pass up a handle to the request here, and pipe through another method where it can be stopped on cancel.